handler/README.md

```diff
@@ -1,232 +1,378 @@
+graphs, we need to understand how nodes are connected.
 
+##### Transitions
+A directed graph consists of nodes and edges. Edges are called transitions. Their configuration looks like:
 ```hocon
-product-body-fallback {
-  factory = "inline-body"
-  config {
-    body = <div>Product not available at the moment</div>
+onTransitions {
+  _success {
+    # next node when _success transition
+  }
+  _error {
+    # next node when error occurs
+  }
+  customTransition {
+    # next node when custom transition
   }
 }
 ```
+Transition is simple text. The `_success` and `_error` transitions are the default ones. However, 
+they are not mandatory!
 
-The default `body` value is empty content.
+>If a node responds with _success transition, but the transition is not configured, then processing is finished.
 
-### Inline Payload Action
-Inline Payload Action puts JSON / JSON Array in Fragment payload with specified key (alias). Its 
-configuration looks like:
+>If a node responds with _error transition, but the transition is not configured, then an exception is returned.
+
+Nodes can declare custom transitions. Custom transitions allow to react to non standard situations 
+such as data sources timeouts, fallbacks etc.
 
+#### Complex example
 ```hocon
-product-payload-fallback {
-  factory = "inline-payload"
-  config {
-    alias = product
-    payload {
-      productId = 1234
-      description = "some description"
+subTasks = [
+  { 
+    action = book-rest-api
+    onTransitions {
+      _success {
+        action = score-algorithm
+      }
+      _error {
+        action = book-from-cache
+      } 
     }
-    # payload = [
-    #   "first product", "second product"
-    # ]
+  },
+  { 
+    action = author-rest-api 
+  }
+]
+onTransitions {
+  _error {
+    action = book-and-author-fallback
   }
 }
 ```
-The default `alias` is action alias.
+The above task fetches book and author details in parallel. The `book-rest-api` action node defines 
+`_success` and `_error` transitions. If any subtask fail then the `book-and-author-fallback` node action 
+is executed. 
 
-### Payload To Body Action
-Payload To Body Action copies to Fragment body specified payload key value. Its configuration looks like:
+## Actions
+Action defines action node logic, it is the `F -> (F',T)` function. Actions integrate with external data sources, 
+do some fragments modifications or fetch data. A data source response is saved in a Fragment's payload (JSON object) 
+under an Action's name key and a "\_result" sub-key:
+```json
+{
+  "book": {
+    "_result": { }
+  }
+}
+```
+
+Actions are divided in two types:
+- `simple actions` that integrate with data sources
+- `behaviours` that wrap simple actions and add some "behaviour"
+
+### Simple Actions
+
+#### HTTP Action
+The HTTP Action fetches JSON data from REST APIs (GET request). See more [here](https://github.com/Knotx/knotx-data-bridge/tree/master/http).
```

Comment:
Could you add some example usage of `http` action?

Reply:
It's already there: https://github.com/Knotx/knotx-data-bridge/tree/master/http#how-to-use

Reply:
Yeah, but we provide small snippets for Inline Body Action, Inline Payload Action and Payload To Body Action but not the HTTP action in this file.